### ISSUE TITLE
incompatible shape in conv1D when using strides as value '2'

### ISSUE THREAD

**Issue by Shan89G** [keras-team/keras#13649]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  1.12
- Keras version:  2.2.5
- Python version:  3.6
- CUDA/cuDNN version:  N?A
- GPU model and memory:  NA

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  
conv1d_output= SeparableConv1D(kernel_size=2, filters=1, padding='same',activation='tanh',input_shape=(None,None),strides=2,kernel_initializer= 'uniform',)(output)
incompatible shape issue in feature set [20,4418] vs [20,2209]
**Describe the expected behavior**  

**Code to reproduce the issue**  
Provide a reproducible test case that is the bare minimum necessary to generate the problem.  

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by Lasers67** [comment]
Could you be more specific by providing the code? It is not possible to estimate, without knowing sizes of input and the other layers,
One Possible issue could be that when estimating the size of padding using formula
op = (ip+2*p-f)/s + 1 where
op is size of output, ip is size of input, p is size of padding, f is kernel size and s is stride

Using stride of 2 may give p as a decimal(when op and ip sizes are same due to 'same' padding)

**Comment by Lasers67** [comment]
Could you be more specific by providing the code? It is not possible to estimate, without knowing sizes of input and info about the other layers,
One Possible issue could be that when estimating the size of padding using formula
op = (ip+2*p-f)/s + 1 where
op is size of output, ip is size of input, p is size of padding, f is kernel size and s is stride

Using stride of 2 may give p as a decimal and result in inconsistency across backends as referred in documentation(when op and ip sizes are same due to 'same' padding)

https://github.com/keras-team/keras/pull/9473#issuecomment-395383922

**Comment by Shan89G** [comment]
> Could you be more specific by providing the code? It is not possible to estimate, without knowing sizes of input and info about the other layers,
> One Possible issue could be that when estimating the size of padding using formula
> op = (ip+2*p-f)/s + 1 where
> op is size of output, ip is size of input, p is size of padding, f is kernel size and s is stride
> 
> Using stride of 2(>1) may give inconsistencies across backends as referred in documentation(when op and ip sizes are same due to 'same' padding)
> 
> Please refer
> 
> [#9473 (comment)](https://github.com/keras-team/keras/pull/9473#issuecomment-395383922)

### Define the model structure
def create_model(x,y,z, R):

    x_input = Input(shape=(None,),dtype='int32',name='words_input')
    x1 = Embedding(input_dim=x.shape[0], output_dim=x.shape[1],  weights=[x], trainable=False)(words_input)
    y_input = Input(shape=(None,), dtype='int32', name='casing_input')
    y1 = Embedding(output_dim=y.shape[1], input_dim=y.shape[0], weights=[y], trainable=False)(casing_input)
    z_input = Input(shape=(None,), dtype='int32', name='newline_input')
    z1 = Embedding(output_dim=z.shape[1], input_dim=z.shape[0], weights=[z], trainable=False)(newline_input)
    output = concatenate([x1, y1,z1])
    conv1d_output= SeparableConv1D(kernel_size=100, filters=82, padding='valid',activation='tanh',strides=1)(output)
    output = Dropout(0.5)(conv1d_output)    	
    flatten_output=TimeDistributed(Flatten())(output)
    output = TimeDistributed(Dense(len(R), activation='softmax'))(flatten_output)
    model = Model(inputs=[x_input, y_input,z_input], outputs=[output])
    model.compile(loss='sparse_categorical_crossentropy', optimizer='nadam',metrics = ['sparse_categorical_accuracy'])

My model.fit()
model.fit([x1, y1,z1], R,verbose=1)

Input shape
(0,0,82)

Sample train input
x y z R (table format)
input size = 4418 (no of lines)


Model should predict R

**Comment by Shan89G** [comment]
Can anyone could clarify this incompatible shape issue? 
is it possible to use strides value as 2 in Conv1D network?
My input length is differ with number of output observation when I used strides as 2. Its like , observation count is half the value of inputs

**Comment by sushreebarsa** [comment]
@Shan89G We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.Thanks!
